### DISCUSSION TITLE
DEFI liquidate：Do we need to burn debtor's DSC?

### DISCUSSION THREAD

**Discussion by serena0323** [Cyfrin/foundry-full-course-cu#3121] [Q&A] [answered]
On our stable coins liquidate function, we burn liquidator's DSC, but I didn't see we burn debtor's DSC, do we need it? I am not sure if I miss something.
'''
function liquidate(
        address collateral,
        address user,
        uint256 debtToCover
    )
        external
        isAllowedToken(collateral)
        moreThanZero(debtToCover)
        nonReentrant
    {
        uint256 startingUserHealthFactor = _healthFactor(user);
        if (startingUserHealthFactor >= MIN_HEALTH_FACTOR) {
            revert DSCEngine__HealthFactorOk();
        }
        // If covering 100 DSC, we need to $100 of collateral
        uint256 tokenAmountFromDebtCovered = getTokenAmountFromUsd(collateral, debtToCover);
        // And give them a 10% bonus
        // So we are giving the liquidator $110 of WETH for 100 DSC
        // We should implement a feature to liquidate in the event the protocol is insolvent
        // And sweep extra amounts into a treasury
        uint256 bonusCollateral = (tokenAmountFromDebtCovered * LIQUIDATION_BONUS) / LIQUIDATION_PRECISION;
        // Burn DSC equal to debtToCover
        // Figure out how much collateral to recover based on how much burnt
        _redeemCollateral(collateral, tokenAmountFromDebtCovered + bonusCollateral, user, msg.sender);
        _burnDsc(debtToCover, user, msg.sender);

        uint256 endingUserHealthFactor = _healthFactor(user);
        // This conditional should never hit, but just in case
        if (endingUserHealthFactor <= startingUserHealthFactor) {
            revert DSCEngine__HealthFactorNotImproved();
        }
        revertIfHealthFactorIsBroken(msg.sender);
    }
    '''

**Comment by EngrPips** [comment]
We liquidated the mean we make them pay more than normal for the DSC they are holding since they are not healthy, So we don't burn their DSC as we don't have it, but we have taken more of their collateral instead. Does this make sense?

**Reply by serena0323** [reply]
Sorry, I am confused. Is the following logic correct:
Supports userA deposited 1000 ETH and minted 500 DSC. Then userA's healthfactor broken. 
So userB help to liquidated. It will transfer some deposited(supports 400 ETH) from userA to userB, and burn userB 300 DSC. And deduct userA's s_DSCMinted amount. So userA's healthfactor will health again。
At this time userA has 600 ETH as deposited, but still has 500 DCS(because when burning, it only burn userB's DSC, but not burn userA's DSC)?

**Reply by EngrPips** [reply, marked as answer]
Yeah, the user holds the DSC they minted, but they have essentially paid for it by we force-taking out of their collateral. The codebase is far from perfect, though, so you might find some edge cases. The codebase intends to teach us the concept of a Stable coin and not build a flawless protocol.

**Reply by serena0323** [reply]
Thanks for your answer. I get it now.

**Reply by EngrPips** [reply]
Anytime Friend. Keep crushing it.
